app/helpers/setting_helper.js

```diff
@@ -109,6 +111,11 @@ const settingHelper = (() => {
     const endpointUrl = EndpointUrl.findByUrlValue(savedEndpointUrl);
     return !!endpointUrl ? endpointUrl.id : null;
   }
+
+  async function getSelectedProposeIndicatorMethod() {
+    const settingData = await getSettingData();
+    return settingData.proposedIndicatorMethod == INDICATOR_BASE ? INDICATOR_BASED_ENUM : PARTICIPANT_BASED_ENUM;
```

Comment:
@limkimsan can we design proposed_indicator_based dictionary s.th like this PROPOSED_INDICATOR_METHODS = {participant_based: {id: 1, display_name: "Participant-based"}, indicator_based: {id: 2, display_name: "Indicator-based"} -> then use this dictionary s.th like this PROPOSED_INDICATOR_METHODS[method]["id"] & PROPOSED_INDICATOR_METHODS[method]["display_name"] so that we will be able to remove the if/else condition

Reply:
Noted, bong. I will make an update on that.

Reply:
if you're based on the naming("indicator_base") 


if you're on id based so your dictionary should be 
PROPOSED_INDICATOR_METHODS = {"1": {name: "participant_based", display_name: "Participant-based"}, "2": {name: "indicator_based", display_name: "Indicator-based"}